aicsimageio/readers/lif_reader.py

```diff
@@ -0,0 +1,768 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import io
+import logging
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple
+
+import dask.array as da
+import numpy as np
+from dask import delayed
+from lxml.etree import Element
+from readlif import utilities
+from readlif.reader import LifFile
+
+from .reader import Reader
+from .. import exceptions, types
+from ..buffer_reader import BufferReader
+from ..constants import Dimensions
+
+###############################################################################
+
+log = logging.getLogger(__name__)
+
+###############################################################################
+
+
+class LifReader(Reader):
+    """
+    LifReader wraps readlif.reader to provide the same reading capabilities but abstracts the specifics of using the
+    backend library to create a unified interface. This enables higher level functions to duck type the File Readers.
+
+    Parameters
+    ----------
+    data: types.FileLike
+        A string or path to the LIF file to be read.
+    chunk_by_dims: List[str]
+        The dimensions to use as the for mapping the chunks / blocks.
+        Default: [Dimensions.SpatialZ, Dimensions.SpatialY, Dimensions.SpatialX]
+        Note: SpatialY and SpatialX will always be added to the list if not present.
+    S: int
+        If the image has different dimensions on any scene from another, the dask array construction will fail.
+        In that case, use this parameter to specify a specific scene to construct a dask array for.
+        Default: 0 (select the first scene)
+    """
+
+    LIF_MAGIC_BYTE = 0x70
+    LIF_MEMORY_BYTE = 0x2a
+
+    ########################################################
+    #
+    #  Note LifFile treats scenes as separate images in the lif file.
+    #  So once a Scene/Image is loaded the image data is retrieved
+    #  2D XY plane by 2D XY plane.
+    #
+    ########################################################
```

Comment:
This makes a bit more sense. But I guess. How is this different than any other file format?

Also, is the data really stored in `XY` or is it `YX`?

Reply:
From the format side it's totally consistent from the reader side this is the first reader I've encountered that reads XY slices instead of some larger dimensional n-gon.

Reply:
Ohhhh I see. The max you can pull out at a time is a single YX plane. I would phrase this comment as:

> Note LifFile (the underlying library of this Reader) only allows for reading a single YX plane at a time, unlike other underlying libraries such as tiffffile or aicspylibczi that allow for reading n-dim chunk reading.